scripts/globals/items/amanomurakumo.lua

```diff
@@ -1,13 +1,85 @@
 -----------------------------------------
--- ID: 18318
+-- ID: 18318, 18319, 18646, 18660, 18674, 19755, 19848, 21015, 21016, 21954
 -- Item: Amanomurakumo
 -- Additional Effect: 10% Attack Down
 -----------------------------------------
 require("scripts/globals/status");
 require("scripts/globals/magic");
 require("scripts/globals/msg");
+require("scripts/globals/weaponskills");
+require("scripts/globals/weaponskillids");
 -----------------------------------
 
+-- https://www.bg-wiki.com/bg/Relic_Aftermath
+local aftermathTable = {};
+
+-- Amanomurakumo 75
+aftermathTable[18318] =
+{
+    power=1,
+    duration = function(tp) return math.floor(0.02 * tp); end,
+    mods =
+    {
+        { id=MOD_STORETP, power=7 }
+    }
+};
+aftermathTable[18319] = aftermathTable[18318]; -- Amanomurakumo (80)
+aftermathTable[18646] = aftermathTable[18318]; -- Amanomurakumo (85)
+aftermathTable[18660] = aftermathTable[18318]; -- Amanomurakumo (90)
+aftermathTable[18674] = aftermathTable[18318]; -- Amanomurakumo (95)
+aftermathTable[19755] = aftermathTable[18318]; -- Amanomurakumo (99)
+aftermathTable[19848] = aftermathTable[18318]; -- Amanomurakumo (99/II)
+aftermathTable[21015] = aftermathTable[18318]; -- Amanomurakumo (119)
+aftermathTable[21016] = aftermathTable[18318]; -- Amanomurakumo (119/II)
+
+-- Amanomurakumo (119/III)
+aftermathTable[21954] =
+{
+    power=2,
+    duration = function(tp) return math.floor(0.06 * tp); end,
+    mods =
+    {
+        { id=MOD_STORETP, power=10 },
+        { id=MOD_ZANSHIN, power=10 }
+    }
+};
+
+function onWeaponskill(user, target, wsid, tp, action)
+    if (wsid == WEAPONSKILL_TACHI_KAITEN) then -- Tachi: Kaiten onry
+        local itemId = user:getWeaponID(SLOT_MAIN);
+        if (aftermathTable[itemId]) then
+            -- Apply the effect and add mods
+            addAftermathEffect(user, tp, aftermathTable[itemId]);
+            -- Add a listener for when aftermath wears (to remove mods)
+            user:addListener("EFFECT_LOSE", "AFTERMATH_LOST", aftermathLost);
+        end
+    end
+end
+
+function aftermathLost(target, effect)
+    if (effect:getType() == EFFECT_AFTERMATH) then
+        local itemId = target:getWeaponID(SLOT_MAIN);
+        if (aftermathTable[itemId]) then
+            -- Remove mods
+            removeAftermathEffect(target, aftermathTable[itemId]);
+            -- Remove the effect listener
+            target:removeListener("AFTERMATH_LOST");
+        end
+    end
+end
+
+function onItemCheck(player, param, caster)
+    if (param == ITEMCHECK_EQUIP) then
+        player:addListener("WEAPONSKILL_USE", "AFTERMATH", onWeaponskill);
```

Comment:
my preference is, unless you're using onWeaponskill more than once, just define it inline as the parameter (and the same thing for aftermathLost)

Reply:
Binding a listener in a listener inline was getting really ugly. What variables to reference where was extremely confusing. This was more readable. Also, aftermathLost gets called when the item is unequipped to make sure the mods are removed from the player.